feat: Introduce session hooks

MoritzWeber0 opened the pull request:
The sessions routes and the kubernetes operator grew a lot in the past. These two files can be hard to understand and code quality was rather bad.

A lot of tool-specific logic happened in the two files, e.g., the creation of a T4C session token.
This logic doesn't belong to the operator or the sessions module, the context is tool-specific, but it has a strong connection to the sessions lifecycle.

The solution is session hooks. For the beginning, we support the following hooks:
- configuration_hook (Configuration for the session)
- post_session_creation_hook (Hook executed after session creation)
- post_session_termination_hook (Hook executed after session termination)

The T4C, pure::variants and Jupyter logic was moved to session-hooks.

This is just a start and should make refactoring of the operator easier in the future.

amolenaar reviewed (approved):
Looks like a fair bit of cleanup. I like it.

Some minor comments, as usual (and to show I actually read it all :wink: )

MoritzWeber0 commented on backend/capellacollab/sessions/hooks/__init__.py:
```diff
@@ -0,0 +1,24 @@
+    return [
+        hook
+        for integration, hook in REGISTERED_HOOKS.items()
+        if integration in tool.integrations.__dict__
+        and tool.integrations.__dict__[integration] is True
```
This is so much cleaner. Is implemented.

MoritzWeber0 commented on backend/capellacollab/sessions/hooks/t4c.py:
```diff
@@ -0,0 +1,124 @@
+
+        return environment, warnings
+
+    def post_session_termination_hook(
+        self,
+        operator: operators.KubernetesOperator,
+        session: sessions_models.DatabaseSession,
+        **kwargs,
+    ):
+        """Hook after session termination
+
+        This hook is executed after the session was terminated by the operator.
+
+        Parameters
+        ----------
+        operator : operators.KubernetesOperator
+            Operator, which is used to spawn the session
+        session_id : str
+            ID of the session
+        """
```
Good catch.

MoritzWeber0 commented on backend/capellacollab/sessions/routes.py:
```diff
@@ -218,25 +209,31 @@ def request_session(
+        ports={"rdp": 3389, "metrics": 9118, "fileservice": 8000},
     )
 
     return create_database_and_guacamole_session(
```
Yes. But I wanted to keep the scope as small as possible, so I'd not do it as part of this PR.

MoritzWeber0 commented:
While looking at it again, I found two issues: 

- The TeamForCapella session token removal was accidentally placed in the `Jupyter`-hook. I moved it to the T4C hook.
- A session was passed to the `post_termination_hook`, which was deleted before. To handle the issue, I changed the `post_termination_hook` to `pre_termination_hook`.
